src/Dialog/Dialog.d.ts

```diff
@@ -10,7 +10,7 @@ export interface DialogProps
   fullWidth?: boolean;
   maxWidth?: 'xs' | 'sm' | 'md' | false;
   transition?: React.ReactType;
-  transitionDuration?: TransitionDuration;
+  transitionDuration?: TransitionTimeout;
```

Comment:
Maybe we should be using the default transition timeout typing: `FadeProps.timeout` for this component.

Reply:
I find having `TransitionDuration` & `TransitionTimeout` confusing. What about having none and using:
- `TransitionProps['timeout']`
- `TransitionProps['timeout'] | 'auto'`